src/Optimizer/Transformer/MinifyHtml.php

```diff
@@ -245,4 +254,39 @@ private function minifyJson(DOMText $node, ErrorCollection $errors)
             $node->data = str_replace(['\u003E', '\u003C'], ['\u003e', '\u003c'], $data);
         }
     }
+
+    /**
+     * Checks if a node is meant to be an inline amp-script.
+     *
+     * @param Element $node The element node need to be checked.
+     * @return bool
+     */
+    private function isInlineAmpScript(Element $node)
+    {
+        $type = $node->getAttribute(Attribute::TYPE);
+        $target = $node->getAttribute(Attribute::TARGET);
+
+        return $type === Attribute::TYPE_TEXT_PLAIN || $target === Protocol::AMP_SCRIPT;
+    }
+
+    /**
+     * Minify inline AMP script.
+     *
+     * @param DOMText         $node   The node to be minified.
+     * @param ErrorCollection $errors Collection of errors that are collected during transformation.
+     */
+    private function minifyAmpScript(DOMText $node, ErrorCollection $errors)
+    {
+        if (! class_exists(Peast::class) || ! class_exists(Renderer::class) || ! class_exists(Compact::class)) {
+            $errors->add(
+                MissingPackage::withMessage('mck89/peast package is required to minify inline amp-script.')
+            );
+            return;
+        }
+
+        $parser = Peast::latest($node->data, [])->parse();
+        $renderer = new Renderer();
+        $renderer->setFormatter(new Compact());
+        $node->data = $renderer->render($parser);
```

Comment:
This should be wrapped in a `try/catch` block, with the `catch` clause adding an error to the `ErrorCollection`. This way, there's a trace if a parsing error happens, but we don't fatal, and we leave the unminified piece of JS in place.